ddtrace/contrib/asyncio/compat.py

```diff
@@ -1,27 +1,15 @@
-import asyncio
-
+from ddtrace.internal.utils.deprecations import DDTraceDeprecationWarning
 from ddtrace.vendor.debtcollector import deprecate
 
+from ..internal.asyncio.compat import *  # noqa: F401,F403
 
-if hasattr(asyncio, "current_task"):
-
-    def asyncio_current_task():
-        deprecate(
-            "ddtrace.contrib.asyncio.create_task(..) is deprecated. The ddtrace library fully supports propagating"
-            " trace contextes to async tasks. No additional configurations are required.",
-            version="3.0.0",
-        )
-        try:
-            return asyncio.current_task()
-        except RuntimeError:
-            return None
 
-else:
+def __getattr__(name):
```

Comment:
We should use `__getattr__` to log a deprecation warning every time this module is accessed.  With this approach the following snippet will log 4 descriptive deprecation warnings:

```
from ddtrace.contrib.asyncio import compat # compat is deprecated
compat.some_func # some_func is deprecated
compat.some_variable # some_variable is deprecared
compat.some.nested_mod # some.nested_mode is deprecated
```